docs/src/index.md

```diff
@@ -54,6 +54,19 @@ or other graphical user interface. It is primarily a programming library, not a
+
+[Universal Algebra](https://en.wikipedia.org/wiki/Universal_algebra) (sometimes called general algebra) is the field of mathematics that studies algebraic structures themselves, not examples ("models") of algebraic structures.
+For instance, rather than take particular groups as the object of study, in universal algebra one takes the class of groups as an object of study. In an algebraic theory, you have a collection of (total) operations and they obey a set of equational axioms. Classically, there is only a single generating type, but there are also typed or multi-sorted versions of algebraic theories. Most of the classical structures of abstract algebra, such as groups, rings, and modules, can be defined as algebraic theories.
+
+Importantly, a category cannot be defined as a algebraic theory. The reason is that the operation of composition is partial, since you can only compose morphisms with compatible (co)domains. Now, categories sure feel like algebraic structures, so people have come up with generalizations of algebraic theories that accomodate categories and related structures.
+
+The first of these was Freyd’s essentially algebraic theories. In an essentially algebraic theory, you can have partially defined operations; however, to maintain the equational character of the system, the domains of operations must themselves be defined equationally. For example, the theory of category would be defined as having two types, Ob and Hom, and the composition operation `compose(f::Hom,g::Hom)::Hom` would have domain given by the equation `codom(f) == dom(g)`. As your theories get more elaborate, the sets of equations defining the domains get more complicated and reasoning about the structure is overwhelming.
+
+Later, Cartmell proposed generalized algebraic theories, which solves the same problem but in a different way. Rather than having partial operations, you have total operations but on dependent types (types that are parameterized by values). So now the composition operation has signature `compose(f::Hom(A,B), g::Hom(B,C))::Hom(A,C) where (A::Ob, B::Ob, C::Ob)`  exactly as appears in Catlab. This is closer to the way that mathematicians actually think and write about categories. For example, if you look at the definitions of category, functor, and natural transformation in Emily Riehl’s textbook, you will see that they are already essentially in the form of a GAT, whereas they require translation into an essentially algebraic theory. Nevertheless, GATs and essentially algebraic theories have the same expressive power, at least in their standard set-based semantics. GATs provide a version of the computer scientist's type theory that plays well with the mathematician's algebra, thus, providing a perfect opportunity for computer algebra systems.
```

Comment:
```suggestion
Nevertheless, GATs and essentially algebraic theories have the same expressive power, at least in their standard set-based semantics.
```
Although this claim (which is originally from Cartmell's thesis) is widely accepted as folklore, it's actually a dubious one. See, for instance, [Voevodsky's comments on the subject](https://www.math.ias.edu/Voevodsky/files/files-annotated/Dropbox/Unfinished_papers/Dynamic_logic/Stage_current/2013_03_05_TTS.pdf). I think it'd be worth simply excluding this comment.

Reply:
Since this is not a research article, I think this is fine. I'll defer to @epatters for his judgement.

Reply:
The claim also appears in Cartmell's 1986 paper with a sketch of a proof. I wasn't aware of Voevodsky's worry, but if I'm reading him correctly, he is worried about whether generalized algebraic and essentially algebraic theories are in 1-1 correspondence, not whether they generate the same class of set-theoretic models (which is the claim asserted above). Thanks for the reference.

Reply:
Ah, you're right; I had misremembered the concern. Cartmell's sketch is a little too unrigorous for my liking, but I should probably just take that as a prompt to write it out in a bit more detail to convince myself :) Thanks.